src/main/java/com/mongodb/kafka/connect/source/MongoSourceTask.java

```diff
@@ -345,6 +349,30 @@ String getTopicNameFromNamespace(final String prefix, final BsonDocument namespa
   }
 
   Map<String, Object> createPartitionMap(final MongoSourceConfig sourceConfig) {
+    return createPartitionMap(sourceConfig, partitionMap == null);
+  }
+
+  private Map<String, Object> createPartitionMap(final MongoSourceConfig sourceConfig, final boolean recreate) {
+    if (recreate) {
+      ConnectionString connectionString = sourceConfig.getConnectionString();
+      StringBuilder builder = new StringBuilder();
+      builder.append(connectionString.isSrvProtocol() ? "mongodb+srv://" : "mongodb://");
+      if (connectionString.getCredential() != null) {
+        builder.append("<username>:<password>@");
```

Comment:
Should we just leave this out entirely?  If we're not distinguishing between different credentials, should we distinguish between the absence or presence of credentials?

Reply:
Sure can do - will remove